app/code/Magento/Customer/Model/ResourceModel/AddressRepository.php

```diff
@@ -139,7 +139,17 @@ public function save(\Magento\Customer\Api\Data\AddressInterface $address)
         $this->addressRegistry->push($addressModel);
         $this->updateAddressCollection($customerModel, $addressModel);
 
-        return $addressModel->getDataModel();
+        $defaultBillingAddressId = null;
+        if ($customerModel->getDefaultBillingAddress() instanceof CustomerAddressModel) {
```

Comment:
@maghamed I see. What about returning Address or false? I would consider this a potential source of errors and it complicates the client code. Why not return Address or null in this case?